records_mover/db/redshift/records_copy.py

```diff
@@ -3,92 +3,121 @@
 from records_mover.records.records_format import (
     BaseRecordsFormat, DelimitedRecordsFormat, AvroRecordsFormat
 )
+from records_mover.records.delimited import ValidatedRecordsHints
 from records_mover.mover_types import _assert_never
 from sqlalchemy_redshift.commands import Format, Encoding, Compression
 from typing import Dict, Optional, Set
 
 RedshiftCopyOptions = Dict[str, Optional[object]]
 
 
-def redshift_copy_options(unhandled_hints: Set[str],
-                          records_format: BaseRecordsFormat,
-                          fail_if_cant_handle_hint: bool,
-                          fail_if_row_invalid: bool,
-                          max_failure_rows: Optional[int]) -> RedshiftCopyOptions:
-    redshift_options: RedshiftCopyOptions = {}
+class RedshiftCopyParamaters:
```

Comment:
What do you think of using a factory abstraction instead? It seems like what we want to do is initialize _something_ with a few settings (fail_if_cant_handle_hint, fail_if_row_invalid, etc), then use that object to also store some state (unhandled_hints, copy_options) as we generate the desired output params.

Basically I think having a top-level object that stores these as `self` vars and has the `process_x` functions as methods could make a lot of sense. Thoughts?

Reply:
I actually don't know that I'm going to write this as a factory pattern but we can look at it afterward and see if we need to refactor again.